python/vision_explanation_methods/error_labeling/error_labeling.py

```diff
@@ -0,0 +1,248 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the Error Labeling Manager class."""
+
+from copy import deepcopy
+from enum import Enum
+
+import numpy as np
+import torchvision
+from torch import Tensor
+
+LABELS = 'labels'
+
+
+class ErrorLabelType(Enum):
+    """
+    Enum providing types of error labels.
+
+    If none, then the detection is not an error. It is a
+    correct prediction.
+    """
+
+    # the gt doesn't have a corresponding detection
+    MISSING = "missing"
+
+    # the model predicted detections, but there was nothing there
+    # this prediction must have a 0 iou score with all gt detections
+    BACKGROUND = "background"
+
+    # the predicted class is correct, bounding box does not have sufficient
+    # overlap with ground truth (based on the iou threshold)
+    LOCALIZATION = "localization"
+
+    # the predicted class is incorrect, the bounding box is correct
+    CLASS_NAME = "class_name"
+
+    # both the predicted class and bounding box are incorrect
+    CLASS_LOCALIZATION = "class_localization"
+
+    # the predicted class is correct, the bounding box is correct, but
+    # the iou score is lower than another detection
+    DUPLICATE_DETECTION = "duplicate_detection"
+
+    MATCH = "match"
+
+
+class ErrorLabeling():
+    """
+    Defines a wrapper class of Error Labeling for vision scenario.
+
+    Only supported for object detection at this point.
+    """
+
+    def __init__(self,
+                 task_type: str,
+                 pred_y: list,
+                 true_y: list,
+                 iou_threshold: float = 0.5):
+        """
+        Create an ErrorLabeling object.
+
+        :param task_type: The task to run.
+        :type task_type: str
+        :param pred_y: predicted detections, nested list of 6 floats (class,
+        bounding box, conf score). The bounding box will be located at indexes
+        1-4.
+        :type pred_y: list
+        :param true_y: ground truth detections, nested list of 6 floats (class,
+        bounding box, is crowded). The bounding box will be located at indexes
+        1-4.
+        :type true_y: list
+        :param iou_threshold: required minimum for bounding box overlap
+        :type iou_threshold: float
+        """
+        self._is_run = False
+        self._is_added = False
+        self._task_type = task_type
+        self._pred_y = pred_y
+        self._true_y = true_y
+        self._iou_threshold = iou_threshold
+        self._match_matrix = np.full((len(self._true_y), len(self._pred_y)),
+                                     None)
+
+    def compute(self, **kwargs):
+        """
+        Compute the error analysis data.
+
+        Note: if a row does not have a match, that means that there is a
+        missing gt detection
+
+        :param kwargs: The keyword arguments to pass to the compute method.
```

Comment:
it looks like this is not a manager but you are still using the compute method and including the kwargs.  I would suggest using a different method name to limit confusion and also removing the kwargs here.